app/server.go

```diff
@@ -1614,3 +1624,142 @@ func (s *Server) HttpService() httpservice.HTTPService {
 func (s *Server) SetLog(l *mlog.Logger) {
 	s.Log = l
 }
+
+func (a *App) GenerateSupportPacket() []model.FileData {
+	// If any errors we come across within this function, we will log it in a warning.txt file so that we know why certain files did not get produced if any
+	var warnings string
+
+	// Creating an array of files that we are going to be adding to our zip file
+	fileDatas := []model.FileData{}
+
+	a.generateSupportPacketYaml(&warnings, &fileDatas)
+	a.createPluginsFile(&warnings, &fileDatas)
+	a.createSanitizedConfigFile(&warnings, &fileDatas)
+	a.getMattermostLog(&warnings, &fileDatas)
+	a.getNotificationsLog(&warnings, &fileDatas)
```

Comment:
Prefer to return warnings and filedatas, keeps the other functions more pure.

Reply:
Good idea! Thanks!